tools/csvstoskyui/csvstoskyui.py

```diff
@@ -0,0 +1,143 @@
+    for key in netboxtopologyfiles:
+        jsontopology = []
+        if not ("show" in netboxtopologyfiles[key]) or not (netboxtopologyfiles[key]["show"]):
+            continue
+
+        csv_reader = csv.DictReader(open(netboxtopologyfiles[key]["filename"]))
+        for idx, raw in enumerate(csv_reader):
+            jsondump = {"Metadata": raw,
+                        "ID": (key + "-" + str(idx)).encode("hex"),
+                        "Origin": "SoftLayer"}
+            jsondump["Metadata"].update({"Type": key})
+            if "valuesfilter" in netboxtopologyfiles[key]:
+                filter_value = False
+                for filterkey, filtervalue in netboxtopologyfiles[key]["valuesfilter"].iteritems():
+                    try:
+                        if not re.search(filtervalue, raw[filterkey]):
+                            filter_value = True
+                    except TypeError:
+                        filter_value = True
+                if filter_value:
+                    continue
+            if "valuesmappingcopy" in netboxtopologyfiles[key]:
+                for mapkey, mapvalue in netboxtopologyfiles[key]["valuesmappingcopy"].iteritems():
+                    jsondump["Metadata"][mapvalue] = jsondump["Metadata"][mapkey]
+            jsontopology.append(jsondump)
+        jsonnodes += jsontopology
+    return jsonnodes
+
+
+## build edges
+def buildEdges(netboxtopologyfiles,jsonnodes):
+    jsonedges = []
+    for key in netboxtopologyfiles:
+        jsontopology = []
+        if not ("show" in netboxtopologyfiles[key]) or not (netboxtopologyfiles[key]["show"]):
+            continue
+
+        if not ("edges" in netboxtopologyfiles[key]) or not (netboxtopologyfiles[key]["edges"]):
+            continue
+
+        edges = netboxtopologyfiles[key]["edges"]
+        for idx, edge in enumerate(edges):
+            relationType = edge["RelationType"]
+            fromtype = key
+            totype = edge["to"].keys()[0]
+
+            fromvalues = edge["from"]
+            tovalues = edge["to"][totype]
+
+            for src in jsonnodes:
+                if src["Metadata"]["Type"] == fromtype:
+                    for dst in jsonnodes:
+                        if dst["Metadata"]["Type"] == totype:
+                            match = True
+                            for idx, fromvalue in enumerate(fromvalues):
+                                tovalue = tovalues[idx]
+                                if tovalue not in dst["Metadata"] or fromvalue not in src["Metadata"] or dst["Metadata"][
+                                    tovalue] != \
+                                        src["Metadata"][fromvalue]:
+                                    match = False
+                            if match:
+                                jsondump = {"Metadata": {"RelationType": relationType},
+                                            "ID": (str("Edge-" + dst["ID"] + src["ID"] + fromtype + totype)).encode("hex"),
```

Comment:
dst["ID"] + src["ID"] is already unique (no need to add types)

Reply:
also not need to encode as hex

Reply:
Thanks, fixed